src/dbnode/storage/flush.go

```diff
@@ -102,28 +105,73 @@ func (m *flushManager) Flush(
 
 	defer m.setState(flushManagerIdle)
 
-	// create flush-er
-	flushPersist, err := m.pm.StartFlushPersist()
```

Comment:
Refactoring out actual flush logic into its own function so that the three loops for flush, compact, and snapshot are clearer.